slider/src/components/SliderPlugin.vue

```diff
@@ -16,44 +29,57 @@ export default {
       default: undefined,
     },
   },
+
   computed: {
+    minValue() {
+      return this.getMinValue()
+    },
+    maxValue() {
+      return this.getMaxValue()
+    },
+    filteredMarks() {
+      const marks = this.getMarks()
+      return this.filterInvalidMarks(marks)
+    },
+    selectedValue() {
+      return this.value?.value ?? this.defaultValue()
+    },
     defaultValue() {
       return this.options.defaultValue
         ? Number.parseFloat(this.options.defaultValue)
         : this.minValue
     },
-    minValue() {
-      return this.options.minValue
-        ? Number.parseFloat(this.options.minValue)
-        : 0
-    },
   },
+
   created() {
-    const defaultValue = parseFloat(this.defaultValue)
     if (!this.value?.value) {
-      this.setValue({ value: defaultValue })
+      this.setValue({ value: this.defaultValue })
     }
   },
-  render() {
-    const minValue = this.minValue
-    const maxValue = this.options.maxValue
-      ? Number.parseFloat(this.options.maxValue)
-      : 100
-    const value = this.value?.value || this.defaultValue
-    const marks =
-      typeof this.options.marks === 'undefined'
+  methods: {
+    getMinValue() {
+      return this.options.minValue
+        ? Number.parseFloat(this.options.minValue)
+        : 0
+    },
+    getMaxValue() {
+      return this.options.maxValue
+        ? Number.parseFloat(this.options.maxValue)
+        : 100
+    },
+    getMarks() {
+      return typeof this.options.marks === 'undefined'
         ? undefined
         : JSON.parse(this.options.marks)
-    return (
-      <HorizontalSlider
-        // archAngle={archAngle}
-        value={value}
-        setValue={(value) => this.setValue({ value })}
-        minValue={minValue}
-        maxValue={maxValue}
-        marks={marks}
-      />
-    )
+    },
+    filterInvalidMarks(marks) {
+      if (!Array.isArray(marks)) {
```

Comment:
Is this checking for `undefined`? Then we could maybe write `marks?.filter(...)` instead and then inline it in `filteredMarks()`.